doc/compiler/aot/SymbolValidationManager.md

```diff
@@ -0,0 +1,173 @@
+
+
+## Subtleties
+
+### Class Chains
+
+Class Chains are a core component of AOT validation; they ensure that
+the shape of a class is the same as that during the compile run. For
+technical details about Class Chains, see 
+[AOT Class Chains](https://github.com/eclipse/openj9/blob/master/doc/compiler/aot/AOTClassChains.md).
+Thus, for every validation record involving classes, a Class Chain
+Validation Record is also created (unless a record for that class
+already exists).
+
+### Array Classes
+
+Array Classes create a problem for validation. Class Chains can only
+exist for non-array classes. However, if the compiler acquired an 
+array class by name or via profiling, in order to use it during an AOT
+compilation, it still needs to ensure that the shape of that class will
+be valid in the load run. Therefore, for every validation record 
+involving classes, what actually happens is the SYM will get the
+leaf component class, store the record as if the compiler acquired the
+leaf component class from the query, and then store as many 
+Array From Component Class Validation Records as there are array
+dimensions. Therefore, on the load run, when performing a validation, 
+the SYM also has ensure to get the leaf component class in order for
+the validation to be consistent with the compile run. The actual array
+class gets validated via the Array From Component Class records.
+
+### Primitive Classes
+
+Primitive Classes create a different problem for validation. As it turns
+out, the ROM Classes for primitive classes aren't stored in the SCC.
+This is an issue when storing Class By Name records, since the name
+portion of the validation is derived from the ROM Class. In order to
+deal with this problem, the Profiled Class and Class By Name 
+Validations also store a char value representing the primitive type;
+if the char value is '\0', then the class being validated isn't
+primitive and the rest of the validation process continues. If the
+char value is not '\0', then class is primitive, and the Profiled 
+Class or Class By Name record returns validaion succeeded, which is
+correct because Primitive Classes are guaranteed to always be the
+same regardless of how they are acquired.
+
+### Guaranteed IDs
+
+Because Primitive Classes (and their associated array classes) are always
+guaranteed to exist, and because the Root Class and method from root class
+have already been validated prior to even attempting to perfom the AOT
+load, these symbols don't need to be validation records stored in 
+the SCC. Therefore, the SYM, in its constructor, initializes the maps
+with these IDs and the associated symbols.
+
+## Benefits
+
+1. Makes explicit the providence of every symbol acquired by the compiler
+2. Enables code paths previously disabled under AOT
+3. Allows relocations of classes/methods associated with cpIndex=-1
+4. Facilitates the enablement of Known Object Table under AOT
+5. Facilitates the enablement of JSR292 under AOT
```

Comment:
Thanks for the descrption at this level documented here. Helps understand the motivations for people reading the PR changes.